### ISSUE TITLE
Unable to build framework

### ISSUE THREAD

**Issue by luibo** [status-im/Keycard.swift#21]
Hi,
I'm trying to build the Keycard framework in order to include it in a project. The problem is that build always fails due to the funcion `secp256k1ECDH` in `Crypto.swift`. The error is:

**Extra arguments at positions #5, #6 in call**

referred to this line of code:

`_ = secp256k1_ecdh(secp256k1Ctx, &ecdhOut, &pubKey, privKey, { (output, x, _, _) -> Int32 in memcpy(output, x, 32); return 1 }, nil)`

This should be a function from the secp256k1.swift framework, and actually the function declaration shows that it needs only 4 arguments. If I delete the last 2 arguments I get another error:

**Undefined symbol: _secp256k1_ecdh**

I have no idea about why this is happening. I used to build the framework in other projects some months ago and I didn't have any problems, this is happening after Xcode and Swift updates. Any suggestion to deal with this?

Thank you.

**Comment by bitgamma** [comment]
I still haven't looked into this in details, but it looks like you got a wrong version of a dependency. The secp256k1 we use is https://github.com/status-im/secp256k1.swift.git where the secp256k1_ecdh function takes 6 arguments (first error you got) and compilation of ECDH is enabled (second error you got). It could be that you fetched the library a long time ago (before the changes were made) and XCode did not update the dependendencies/cache. I very vaguely remember having had this problem too but I don't remember what I did to solve it. Googling I found this https://gist.github.com/maciekish/66b6deaa7bc979d0a16c50784e16d697 but I don't know if it is still current.
